tikv-client/src/main/java/com/pingcap/tikv/TiBatchWriteUtils.java

```diff
@@ -38,18 +37,9 @@
       RowKey endRowKey = RowKey.createBeyondMax(table.getId());
 
       while (key.compareTo(endRowKey) < 0) {
-        System.out.println(KeyUtils.formatBytes(key.toByteString()));
         TiRegion region = session.getRegionManager().getRegionByKey(key.toByteString());
-        if (region == null) {
-          break;
-        } else {
-          regionList.add(region);
-          byte[] endKey = region.getEndKey().toByteArray();
-          if (endKey.length == 0) {
-            break;
-          }
-          key = Key.toRawKey(endKey);
-        }
+        regionList.add(region);
+        key = Key.toRawKey(region.getEndKey());
```

Comment:
I think we an extract this while loop as `getRegionListByKeyRange(startKey, endKey)`, and please notice the situation where `endKey` is empty while `region.endkey` is not.